network/src/sync/master.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{collections::HashMap, net::SocketAddr, time::Duration};
+
+use crate::{NetworkError, Node, Payload, Peer};
+use futures::{pin_mut, select, FutureExt};
+use rand::prelude::SliceRandom;
+use snarkvm_algorithms::crh::double_sha256;
+use snarkvm_objects::{BlockHeader, BlockHeaderHash, Storage};
+use tokio::{sync::mpsc, time::Instant};
+
+pub enum SyncInbound {
+    BlockHashes(SocketAddr, Vec<BlockHeaderHash>),
+    Block(SocketAddr, Vec<u8>),
+}
+
+pub struct SyncMaster<S: Storage + Send + Sync + 'static> {
+    node: Node<S>,
+    incoming: mpsc::Receiver<SyncInbound>,
+}
+
+struct SyncBlock {
+    address: SocketAddr,
+    block: Vec<u8>,
+}
+
+impl<S: Storage + Send + Sync + 'static> SyncMaster<S> {
+    pub fn new(node: Node<S>) -> (Self, mpsc::Sender<SyncInbound>) {
+        let (sender, receiver) = mpsc::channel(256);
```

Comment:
```suggestion
        let (sender, receiver) = mpsc::channel(crate::MAX_BLOCK_SYNC_COUNT);
```

Reply:
We can receive more than 64 blocks since some peers will send us different blocks. Also doesn't hurt to have buffer (basically free)